### ISSUE TITLE
Pip install tensorflow==2.7.0 expects cuda 11.0 instead 11.2

### ISSUE THREAD

**Issue by artus-LYTiQ** [tensorflow/tensorflow#52981]
<em>Please make sure that this is a build/installation issue. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:build_template</em>
Well, it seems to fall between a bug and an installation issue :P

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): ubuntu 20.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip 
- TensorFlow version: 2.6.1 upgrading to 2.7 via pip
- Python version:  3.8.12
- Installed using virtualenv? pip? conda?: conda for env incl cudatoolkit and cudnn, pip for tf 2.6.1 and 2.7.0
- Bazel version (if compiling from source): n/a
- GCC/Compiler version (if compiling from source): n/a
- CUDA/cuDNN version: 11.2.2               he111cf0_8    nvidia; 8.1.0.77             h90431f1_0    conda-forge
- GPU model and memory: k80

**Describe the problem**
Just set up a running TF 2.6.1 using pip in a conda env last week following the requirements as state in https://www.tensorflow.org/install/gpu. When I upgraded to 2.7 via pip the installed package suddenly required cuda 11.0 and thus fails to load/use the gpu. 

**Provide the exact sequence of commands / steps that you executed before running into the problem**
conda update --force conda
conda create -n tf2 python=3.8
conda activate tf2
conda install cudatoolkit==11.3
conda install cudatoolkit==11.3.1
conda install cudnn
conda install pandas numpy scikit-learn
conda install cupti
conda remove cudnn
conda remove cupti
conda remove cudatoolkit
conda install -c nvidia cudatoolkit==11.2
conda install -c nvidia cudatoolkit==11.2.2
conda list -n tf2
conda install -c nvidia cudnn=8.1.0
conda install -c conda-forge cudnn=8.1.0
pip install --upgrade tensorflow     # this was under 2.6.1; needs ==2.6.1 now
conda install matplotlib seaborn
nvidia-smi
conda install -c conda-forge notebook jupyter ipykernel jupyter_contrib_nbextensions
jupyter notebook --generate-config
vi .jupyter/jupyter_notebook_config.py
jupyter notebook password
vi .jupyter/jupyter_notebook_config.py
sudo
sudo print
sudo systemctl start jupyter.service
sudo systemctl stop jupyter.service
sudo systemctl start jupyter.service
pip install tensorflow-datasets
<em>this was my working 2.6.1 install with gpu available</em> Then I proceeded to upgrade to 2.7 once it became available:
pip install -U tensorflow
<em> 
This then leads to the new TF looking for Cuda 11.0!
Installing collected packages: tensorflow-estimator, keras, tensorflow
  Attempting uninstall: tensorflow-estimator
    Found existing installation: tensorflow-estimator 2.6.0
    Uninstalling tensorflow-estimator-2.6.0:
      Successfully uninstalled tensorflow-estimator-2.6.0
  Attempting uninstall: keras
    Found existing installation: keras 2.6.0
    Uninstalling keras-2.6.0:
      Successfully uninstalled keras-2.6.0
  Attempting uninstall: tensorflow
    Found existing installation: tensorflow 2.6.1
    Uninstalling tensorflow-2.6.1:
      Successfully uninstalled tensorflow-2.6.1
Successfully installed keras-2.7.0 tensorflow-2.7.0 tensorflow-estimator-2.7.0

>>> import tensorflow as tf
2021-11-08 09:33:10.345741: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
2021-11-08 09:33:10.345774: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.

**Any other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
# packages in environment at /home/student/miniconda3/envs/tf2:
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             4.5                       1_gnu
absl-py                   0.12.0                   pypi_0    pypi
argon2-cffi               20.1.0           py38h27cfd23_1
astunparse                1.6.3                    pypi_0    pypi
async_generator           1.10                       py_0    conda-forge
attrs                     21.2.0             pyhd8ed1ab_0    conda-forge
backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
backports                 1.0                        py_2    conda-forge
backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
blas                      1.0                         mkl
bleach                    4.1.0              pyhd8ed1ab_0    conda-forge
bottleneck                1.3.2            py38heb32a55_1
brotli                    1.0.9                he6710b0_2
ca-certificates           2021.10.8            ha878542_0    conda-forge
cachetools                4.2.4                    pypi_0    pypi
certifi                   2021.10.8        py38h578d9bd_1    conda-forge
cffi                      1.14.6           py38ha65f79e_0    conda-forge
charset-normalizer        2.0.7                    pypi_0    pypi
clang                     5.0                      pypi_0    pypi
cudatoolkit               11.2.2               he111cf0_8    nvidia
cudnn                     8.1.0.77             h90431f1_0    conda-forge
cycler                    0.10.0                   py38_0
dbus                      1.13.18              hb2f20db_0
debugpy                   1.4.1            py38h709712a_0    conda-forge
decorator                 5.1.0              pyhd8ed1ab_0    conda-forge
defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
dill                      0.3.4                    pypi_0    pypi
entrypoints               0.3             pyhd8ed1ab_1003    conda-forge
expat                     2.4.1                h2531618_2
flatbuffers               1.12                     pypi_0    pypi
fontconfig                2.13.1               h6c09931_0
fonttools                 4.25.0             pyhd3eb1b0_0
freetype                  2.11.0               h70c0345_0
future                    0.18.2                   pypi_0    pypi
gast                      0.4.0                    pypi_0    pypi
giflib                    5.2.1                h7b6447c_0
glib                      2.69.1               h5202010_0
google-auth               2.3.3                    pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
googleapis-common-protos  1.53.0                   pypi_0    pypi
grpcio                    1.41.1                   pypi_0    pypi
gst-plugins-base          1.14.0               h8213a91_2
gstreamer                 1.14.0               h28cd5cc_2
h5py                      3.1.0                    pypi_0    pypi
icu                       58.2                 he6710b0_3
idna                      3.3                      pypi_0    pypi
importlib-metadata        4.8.1            py38h578d9bd_1    conda-forge
importlib-resources       5.4.0                    pypi_0    pypi
intel-openmp              2021.4.0          h06a4308_3561
ipykernel                 6.4.2            py38he5a9106_0    conda-forge
ipython                   7.29.0           py38he5a9106_1    conda-forge
ipython_genutils          0.2.0                      py_1    conda-forge
ipywidgets                7.6.5              pyhd8ed1ab_0    conda-forge
jedi                      0.18.0           py38h578d9bd_3    conda-forge
jinja2                    3.0.2              pyhd8ed1ab_0    conda-forge
joblib                    1.1.0              pyhd3eb1b0_0
jpeg                      9d                   h7f8727e_0
jsonschema                4.1.2              pyhd8ed1ab_0    conda-forge
jupyter                   1.0.0            py38h578d9bd_6    conda-forge
jupyter_client            7.0.6              pyhd8ed1ab_0    conda-forge
jupyter_console           6.4.0              pyhd8ed1ab_0    conda-forge
jupyter_contrib_core      0.3.3                      py_2    conda-forge
jupyter_contrib_nbextensions 0.5.1              pyhd8ed1ab_2    conda-forge
jupyter_core              4.9.1            py38h578d9bd_0    conda-forge
jupyter_highlight_selected_word 0.2.0           py38h578d9bd_1002    conda-forge
jupyter_latex_envs        1.4.6           pyhd8ed1ab_1002    conda-forge
jupyter_nbextensions_configurator 0.4.1            py38h578d9bd_2    conda-forge
jupyterlab_pygments       0.1.2              pyh9f0ad1d_0    conda-forge
jupyterlab_widgets        1.0.2              pyhd8ed1ab_0    conda-forge
keras                     2.6.0                    pypi_0    pypi
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.3.1            py38h2531618_0
lcms2                     2.12                 h3be6417_0
ld_impl_linux-64          2.35.1               h7274673_9
libclang                  12.0.0                   pypi_0    pypi
libffi                    3.3                  he6710b0_2
libgcc-ng                 9.3.0               h5101ec6_17
libgfortran-ng            7.5.0               ha8ba4b0_17
libgfortran4              7.5.0               ha8ba4b0_17
libgomp                   9.3.0               h5101ec6_17
libpng                    1.6.37               hbc83047_0
libsodium                 1.0.18               h36c2ea0_1    conda-forge
libstdcxx-ng              9.3.0               hd4cf53a_17
libtiff                   4.2.0                h85742a9_0
libuuid                   1.0.3                h7f8727e_2
libwebp                   1.2.0                h89dd481_0
libwebp-base              1.2.0                h27cfd23_0
libxcb                    1.14                 h7b6447c_0
libxml2                   2.9.12               h03d6c58_0
libxslt                   1.1.34               hc22bd24_0
lxml                      4.6.3            py38hf1fe3a4_0    conda-forge
lz4-c                     1.9.3                h295c915_1
markdown                  3.3.4                    pypi_0    pypi
markupsafe                2.0.1            py38h497a2fe_0    conda-forge
matplotlib                3.4.3            py38h06a4308_0
matplotlib-base           3.4.3            py38hbbc1b5f_0
matplotlib-inline         0.1.3              pyhd8ed1ab_0    conda-forge
mistune                   0.8.4           py38h497a2fe_1004    conda-forge
mkl                       2021.4.0           h06a4308_640
mkl-service               2.4.0            py38h7f8727e_0
mkl_fft                   1.3.1            py38hd3c417c_0
mkl_random                1.2.2            py38h51133e4_0
munkres                   1.1.4                      py_0
nbclient                  0.5.4              pyhd8ed1ab_0    conda-forge
nbconvert                 6.2.0            py38h578d9bd_0    conda-forge
nbformat                  5.1.3              pyhd8ed1ab_0    conda-forge
ncurses                   6.3                  h7f8727e_0
nest-asyncio              1.5.1              pyhd8ed1ab_0    conda-forge
notebook                  6.4.5              pyha770c72_0    conda-forge
numexpr                   2.7.3            py38h22e1b3c_1
numpy                     1.19.5                   pypi_0    pypi
oauthlib                  3.1.1                    pypi_0    pypi
olefile                   0.46               pyhd3eb1b0_0
openssl                   1.1.1l               h7f8727e_0
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.0               pyhd8ed1ab_0    conda-forge
pandas                    1.3.4            py38h8c16a72_0
pandoc                    2.16                 h7f98852_0    conda-forge
pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
parso                     0.8.2              pyhd8ed1ab_0    conda-forge
pcre                      8.45                 h295c915_0
pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
pickleshare               0.7.5                   py_1003    conda-forge
pillow                    8.4.0            py38h5aabda8_0
pip                       21.2.4           py38h06a4308_0
prometheus_client         0.12.0             pyhd8ed1ab_0    conda-forge
promise                   2.3                      pypi_0    pypi
prompt-toolkit            3.0.21             pyha770c72_0    conda-forge
prompt_toolkit            3.0.21               hd8ed1ab_0    conda-forge
protobuf                  3.19.1                   pypi_0    pypi
ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pygments                  2.10.0             pyhd8ed1ab_0    conda-forge
pyparsing                 3.0.4              pyhd3eb1b0_0
pyqt                      5.9.2            py38h05f1152_4
pyrsistent                0.17.3           py38h497a2fe_2    conda-forge
python                    3.8.12               h12debd9_0
python-dateutil           2.8.2              pyhd3eb1b0_0
python_abi                3.8                      2_cp38    conda-forge
pytz                      2021.3             pyhd3eb1b0_0
pyyaml                    5.4.1            py38h497a2fe_0    conda-forge
pyzmq                     19.0.2           py38ha71036d_2    conda-forge
qt                        5.9.7                h5867ecd_1
qtconsole                 5.1.1              pyhd8ed1ab_0    conda-forge
qtpy                      1.11.2             pyhd8ed1ab_0    conda-forge
readline                  8.1                  h27cfd23_0
requests                  2.26.0                   pypi_0    pypi
requests-oauthlib         1.3.0                    pypi_0    pypi
rsa                       4.7.2                    pypi_0    pypi
scikit-learn              1.0.1            py38h51133e4_0
scipy                     1.7.1            py38h292c36d_2
seaborn                   0.11.2             pyhd3eb1b0_0
send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
setuptools                58.0.4           py38h06a4308_0
sip                       4.19.13          py38he6710b0_0
six                       1.15.0                   pypi_0    pypi
sqlite                    3.36.0               hc218d9a_0
tensorboard               2.7.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.0                    pypi_0    pypi
tensorflow                2.6.1                    pypi_0    pypi
tensorflow-datasets       4.4.0                    pypi_0    pypi
tensorflow-estimator      2.6.0                    pypi_0    pypi
tensorflow-io-gcs-filesystem 0.21.0                   pypi_0    pypi
tensorflow-metadata       1.4.0                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
terminado                 0.12.1           py38h578d9bd_0    conda-forge
testpath                  0.5.0              pyhd8ed1ab_0    conda-forge
threadpoolctl             2.2.0              pyh0d69192_0
tk                        8.6.11               h1ccaba5_0
tornado                   6.1              py38h27cfd23_0
tqdm                      4.62.3                   pypi_0    pypi
traitlets                 5.1.1              pyhd8ed1ab_0    conda-forge
typing-extensions         3.7.4.3                  pypi_0    pypi
urllib3                   1.26.7                   pypi_0    pypi
wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
webencodings              0.5.1                      py_1    conda-forge
werkzeug                  2.0.2                    pypi_0    pypi
wheel                     0.37.0             pyhd3eb1b0_1
widgetsnbextension        3.5.2            py38h578d9bd_0    conda-forge
wrapt                     1.12.1                   pypi_0    pypi
xz                        5.2.5                h7b6447c_0
yaml                      0.2.5                h516909a_0    conda-forge
zeromq                    4.3.4                h9c3ff4c_0    conda-forge
zipp                      3.6.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11               h7b6447c_3
zstd                      1.4.9                haebb681_0

**Comment by artus-LYTiQ** [comment]
Downgrading to 2.6.2 (or 2.6.1 for tf plus manual downgrade for keras to 2.6.0) then brings the system back to a working state:

pip install -U tensorflow==2.6.2
Successfully installed google-auth-1.35.0 keras-2.6.0 tensorboard-2.6.0 tensorflow-2.6.2 tensorflow-estimator-2.6.0
(tf2) student@c4d-vw-14:~$ python
Python 3.8.12 (default, Oct 12 2021, 13:49:34)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import tensorflow as tf
>>> tf.test.is_gpu_available()
WARNING:tensorflow:From <stdin>:1: is_gpu_available (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.config.list_physical_devices('GPU')` instead.
2021-11-08 09:37:09.767605: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX2 FMA
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2021-11-08 09:37:16.173199: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1510] Created device /device:GPU:0 with 10801 MB memory:  -> device: 0, name: Tesla K80, pci bus id: 0001:00:00.0, compute capability: 3.7
True
>>> sys_details = tf.sysconfig.get_build_info()
>>>
>>> cuda_version = sys_details["cuda_version"]
>>>
>>> print(cuda_version)
11.2

Please note that it is not a problem to install cuda 11.0 for tf 2.7 but I wondering why I'd need to do that. As per the official website any version >= 2.5 should support 11.2. I suspect a bug with the pip package.

**Comment by tilakrayal** [comment]
@artus-LYTiQ ,
We can see that you have installed tensorflow from conda environment.Installation issues within the Anaconda environment are tracked in the Anaconda repo.Please try to install in new virtual environment from this [link](https://github.com/tensorflow/tensorflow/releases) and let us know if it is still an issue.Thanks!

**Comment by filippocastelli** [comment]
> @artus-LYTiQ , We can see that you have installed tensorflow from conda environment.Installation issues within the Anaconda environment are tracked in the Anaconda repo.Please try to install in new virtual environment from this [link](https://github.com/tensorflow/tensorflow/releases) and let us know if it is still an issue.Thanks!

I'm observing the same problem in anaconda, downgrading to 2.6 fixes the problem
